src/Orleans/Logging/LoggerWrapper.cs

```diff
@@ -0,0 +1,111 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading;
+
+namespace Orleans.Runtime
+{
+    //TODO: Mark it as [Obsolete] and internal after all runtime has migrated
+    public class LoggerWrapper<T> : Logger
+    {
+        private readonly LoggerWrapper logger;
+        public LoggerWrapper(ILoggerFactory loggerFactory)
+        {
+            logger = new LoggerWrapper(typeof(T).FullName, loggerFactory);
+        }
+
+        public override Severity SeverityLevel => this.logger.SeverityLevel;
+
+        public override string Name => this.logger.Name;
+
+        public override Logger GetLogger(string loggerName)
+        {
+            return this.logger.GetLogger(loggerName);
+        }
+
+        public override void Log(int errorCode, Severity sev, string format, object[] args, Exception exception)
+        {
+            this.logger.Log(errorCode, sev, format, args, exception);
+        }
+    }
+
+    //TODO: Mark it as [Obsolete] and internal after all runtime has migrated
+    public class LoggerWrapper : Logger
```

Comment:
This could be internal today, if the onlt wat to create one of these is through an extension method from loggingFactory, or via DI using `LegacyLogger<T>` (currently `LegacyWrapper<T>`)

Reply:
yes it can be internal eventually. But right now all extensions projects are using LoggerWrapper too. So this cannot be internal in this PR. If all extensions projects moved away from Logger and LoggerWrapper. then this can be marked as internal

Reply:
Right, but if we use the extension method approach (or `Logger<T>` in case of DI), then this wouldn't even need to exist today as a public API.